tests/core/test_basics.py

```diff
@@ -139,6 +139,40 @@ def expected_response_type() -> Type[DummyResponse]:
     assert stream_counter == 2
 
 
+@pytest.mark.asyncio
+async def test_stream_can_get_cancelled(endpoint: Endpoint) -> None:
+    stream_counter = 0
+
+    async_generator = endpoint.stream(DummyRequest)
```

Comment:
If we want to be able to cancel our stream, we have to save a reference to the returned async generator (instead of immediately `async for`'ing over it). We can then later say `await async_generator.aclose()`